### PR TITLE
fix: resolvable scalars not working with asserts

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

Resolvable scalars were not working with assertions.

## Reference

Fixes https://github.com/Redocly/openapi-cli/issues/645

## Testing

TODO:

- [x] add e2e test

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Review by ohorbachevskyi — APPROVED** [review]

**Review by AntonKozachuk — APPROVED** [review]

File: packages/core/src/rules/common/assertions/utils.ts [hunk 1]
```diff
@@ -102,7 +103,9 @@ export function buildSubjectVisitor(
     for (const assert of asserts) {
       if (properties) {
         for (const property of properties) {
-          runAssertion(node[property], assert, location.child(property), report);
+          // we can have resolvable scalar so need to resolve value here.
+          const value = isRef(node[property]) ? resolve(node[property])?.node : node[property];
```
Comment (adamaltman) [inline packages/core/src/rules/common/assertions/utils.ts:107]: I have an open question if we can detect that it is a `$ref`? For example, enforce `$ref` usage for path items (or anything else that you may want to enforce it with). So in other words, I would like to be able to assert `isRef: true`.

Comment (adamaltman) [reply packages/core/src/rules/common/assertions/utils.ts:107]: Also, I would like to assert some things about the ref, like if it is a ref to a another file (as opposed to the components section of the same file).

Comment (RomanHotsiy) [reply packages/core/src/rules/common/assertions/utils.ts:107]: > I would like to be able to assert isRef: true.

We can do it I think. Needs some changes but it should be possible.

> I would like to assert some things about the ref, like if it is a ref to a another file (as opposed to the components section of the same file).

I think we need some syntax idea there then.

Comment (adamaltman) [reply packages/core/src/rules/common/assertions/utils.ts:107]: I made a separate issue for it: https://github.com/Redocly/openapi-cli/issues/648

**Review by adamaltman — APPROVED** [review]

**Pull request merged**